src/import/YAMLConfiguration.ts

```diff
@@ -0,0 +1,380 @@
+import {
+  UsageContext,
+  ImplementationGuideDefinitionTemplate,
+  Meta,
+  Narrative,
+  ImplementationGuideDefinitionResource,
+  ImplementationGuideStatus,
+  ContactPoint,
+  ContactDetail,
+  ImplementationGuideDefinitionPage,
+  ImplementationGuideDefinitionPageGeneration,
+  Quantity,
+  ImplementationGuide,
+  ImplementationGuideDependsOn,
+  ImplementationGuideGlobal,
+  ImplementationGuideDefinitionGrouping,
+  Reference,
+  Identifier
+} from '../fhirtypes';
+
+/**
+ * RawConfiguration follows the proposed configuration format for FSH and incorporates aspects
+ * of the ImplementationGuide, ig.ini, package.json, package-list.json, and menu.xml formats.
+ *
+ * This format is intended to be represented using YAML but is transformed into JSON for our use.
+ *
+ * @see {@link http://hl7.org/fhir/R4/implementationguide.html}
+ * @see {@link https://build.fhir.org/ig/FHIR/ig-guidance/using-templates.html#igroot}
+ * @see {@link https://confluence.hl7.org/display/FHIR/NPM+Package+Specification}
+ * @see {@link https://confluence.hl7.org/pages/viewpage.action?pageId=66928420#FHIRIGPackageListdoco-PublicationObject}
+ * @see {@link https://github.com/FHIR/sample-ig/blob/master/input/includes/menu.xml}
+ */
+export type YAMLConfiguration = {
+  id: ImplementationGuide['id']; // string
```

Comment:
I took this approach of specifying the YAML config type using references to the corresponding ImplemantionGuide properties to ensure closer alignment w/ the related FHIR type.  This wasn't possible everywhere, but where it was possible, I tried to do it.